diff -ur a/rgbe.h b/rgbe.h
--- a/rgbe.h	2014-02-17 23:04:42.000000000 +0100
+++ b/rgbe.h	2017-12-23 11:33:44.944170647 +0100
@@ -62,8 +62,8 @@
             size_t num_scanlines);
 
 
-INLINE void float2rgbe(unsigned char rgbe[4], float red, float green, float blue);
-INLINE void rgbe2float(float *red, float *green, float *blue, unsigned char rgbe[4]);
+void float2rgbe(unsigned char rgbe[4], float red, float green, float blue);
+void rgbe2float(float *red, float *green, float *blue, unsigned char rgbe[4]);
 
 #ifdef __cplusplus
 }
